API Deprecate using labels in bytes format (#18555)
scikit-learn/scikit-learn: doc/whats_new/v1.3.rst

  :pr:`26082` by :user:`Jérémie du Boisberranger <jeremiedbb>` and
  :user:`Olivier Grisel <ogrisel>`.

> - |API| For classifiers and classification metrics, labels encoded as bytes
>   is deprecated and will raise an error in v1.5.
> 
>   :pr:`19664` by :user:`Kaushik Amar Das <cozek>`.

Reviewer: ```suggestion
- |API| For classifiers and classification metrics, labels encoded as bytes
  is deprecated and will raise an error in v1.5.
  :pr:`19664` by :user:`Kaushik Amar Das <cozek>`.
```

  :pr:`26082` by :user:`Jérémie du Boisberranger <jeremiedbb>` and
  :user:`Olivier Grisel <ogrisel>`.

- |API| For classifiers and classification metrics, labels encoded as bytes
  is deprecated and will raise an error in v1.5.

>   :pr:`19664` by :user:`Kaushik Amar Das <cozek>`.

Reviewer: ```suggestion
  :pr:`18555` by :user:`Kaushik Amar Das <cozek>`.
```

  :pr:`26082` by :user:`Jérémie du Boisberranger <jeremiedbb>` and
  :user:`Olivier Grisel <ogrisel>`.

- |API| For classifiers and classification metrics, labels encoded as bytes

>   is deprecated and will raise an error in v1.5.

Reviewer: This needs to be moved to 1.4 and updated to 1.6:

```suggestion
  is deprecated and will raise an error in v1.6.
```

  accepts sparse metrics.
  :pr:`19664` by :user:`Kaushik Amar Das <cozek>`.

:mod:`sklearn.multiclass`
..............................

> - |Enhancement| Raise informative warning message in :func:`type_of_target` when
>   represented as bytes.

Reviewer: This is a API change because it requires users to update their code to remove the warning:

```suggestion
- |API| Raise informative warning message in :func:`type_of_target` when
  represented as bytes. For classifiers and classification metrics, labels encoded 
  as bytes is deprecated and will raise an error in v1.5.
```

  accepts sparse metrics.
  :pr:`19664` by :user:`Kaushik Amar Das <cozek>`.

:mod:`sklearn.multiclass`
..............................

- |API| Raise informative warning message in :func:`type_of_target` when
  represented as bytes. For classifiers and classification metrics, labels encoded 
  as bytes is deprecated and will raise an error in v1.5.

>   :pr:`19664` by :user:`Kaushik Amar Das <cozek>`.

Reviewer: ```suggestion
  :pr:`18555` by :user:`Kaushik Amar Das <cozek>`.
```

  :meth:`semi_supervised.LabelPropagation.fit` now accepts sparse metrics.
  :pr:`19664` by :user:`Kaushik Amar Das <cozek>`.

> :mod:`sklearn.multiclass`

Reviewer: Same here regarding moving to 1.6
